design/concurrent-tree-editing.md

```diff
@@ -0,0 +1,168 @@
+---
+title: concurrent-tree-editing
+target-version: 0.4.6
+---
+
+# Concurrent Tree Editing
+
+## Summary
+
+In Yorkie, users can create and edit JSON-like documents using JSON-like data structures such as `Primitive`, `Object`, `Array`, `Text`, and `Tree`. Among these, the `Tree` structure is used to represent the document model of a tree-based text editor, similar to XML.
+
+This document introduces the `Tree` data structure, and explains the operations provided by `Tree`, focusing on the `Tree` coordinate system and the logic of the `Tree.Edit` operation. Furthermore, it explains how this logic ensures eventual consistency in concurrent document editing scenarios.
+
+### Goals
+
+This document aims to help new SDK contributors understand the overall `Tree` data structure and explain how Yorkie ensures consistency when multiple clients are editing concurrently.
+
+### Non-Goals
+
+This document focuses on `Tree.Edit` operations rather than `Tree.Style`.
+
+## Proposal Details
+
+### JSON-like Tree
```

Comment:
It would be better to express it as an `XML-like Tree` because we use terminology such as `element`, `open tag`, `close tag`, and so on.

```suggestion
### XML-like Tree
```